src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -117,6 +197,53 @@ public static SimilarityFunction fromString(String similarityFunction) {
     }
   }
 
+  /** Collection Id Type enum, default to UUID */
+  public enum IdType {
+    OBJECT_ID,
+    UUID,
+    UUID_V6,
+    UUID_V7,
+    UNDEFINED;
+
+    public static IdType fromString(String idType) {
+      if (idType == null) return UNDEFINED;
+      return switch (idType) {
+        case "objectId" -> OBJECT_ID;
+        case "uuid" -> UUID;
+        case "uuidv6" -> UUID_V6;
+        case "uuidv7" -> UUID_V7;
+        default -> throw ErrorCode.INVALID_ID_TYPE.toApiException(idType);
+      };
+    }
+
+    public String toString() {
+      return switch (this) {
+        case OBJECT_ID -> "objectId";
+        case UUID, UNDEFINED -> "uuid";
```

Comment:
I think "UNDEFINED" should probably become "" since its behavior is slightly different from UUID's (it will result in "plain", unwrapped UUID key)